pkg/generate/app/pipeline.go

```diff
@@ -64,7 +64,7 @@ func (pb *pipelineBuilder) NewBuildPipeline(from string, resolvedMatch *Componen
 			return nil, fmt.Errorf("can't build %q: %v", from, err)
 		}
 		input = inputImage
-		if !input.AsImageStream {
+		if !input.AsImageStream && resolvedMatch.Value != "scratch" {
```

Comment:
This is ugly - it means we're not properly handling !AsImageStream up above.  I would prefer to put something on the match that says this cannot be used as an input image stream